src/acl_usm.cpp

```diff
@@ -380,6 +394,7 @@ clSharedMemAllocINTEL(cl_context context, cl_device_id device,
   // The end of the properties list is specified with a zero.
   cl_mem_alloc_flags_intel alloc_flags = 0;
   std::unordered_set<unsigned long long> seen_flags;
+  cl_int mem_id = -1;
```

Comment:
Would `std::optional<cl_int>` work here? To prevent future mistakes in using an uninitialized value.